scripts/build_cost_data.py

```diff
@@ -0,0 +1,197 @@
+    costs.at["CCGT", "fuel"] = costs.at["gas", "fuel"]
+
+    costs["marginal_cost"] = costs["VOM"] + costs["fuel"] / costs["efficiency"]
+
+    costs.at["OCGT", "CO2 intensity"] = costs.at["gas", "CO2 intensity"]
+    costs.at["CCGT", "CO2 intensity"] = costs.at["gas", "CO2 intensity"]
+
+    costs.at["solar", "capital_cost"] = costs.at["solar-utility", "capital_cost"]
+    costs = costs.rename({"solar-utility single-axis tracking": "solar-hsat"})
+
+    costs = costs.rename(columns={"standing losses": "standing_losses"})
+
+    # Calculate storage costs if max_hours is provided
+    if max_hours is not None:
+
+        def costs_for_storage(store, link1, link2=None, max_hours=1.0):
+            capital_cost = link1["capital_cost"] + max_hours * store["capital_cost"]
+            if link2 is not None:
+                capital_cost += link2["capital_cost"]
+            return pd.Series(
+                {
+                    "capital_cost": capital_cost,
+                    "marginal_cost": 0.0,
+                    "CO2 intensity": 0.0,
+                    "standing_losses": 0.0,
+                }
+            )
+
+        costs.loc["battery"] = costs_for_storage(
+            costs.loc["battery storage"],
+            costs.loc["battery inverter"],
+            max_hours=max_hours["battery"],
+        )
+        costs.loc["H2"] = costs_for_storage(
+            costs.loc["hydrogen storage underground"],
+            costs.loc["fuel cell"],
+            costs.loc["electrolysis"],
+            max_hours=max_hours["H2"],
+        )
+
+    for attr in ("marginal_cost", "capital_cost"):
+        overwrites = config["overwrites"].get(attr)
+        if overwrites is not None:
+            overwrites = pd.Series(overwrites)
+            idx = overwrites.index.intersection(costs.index)
+            costs.loc[idx, attr] = overwrites.loc[idx]
+            warnings.warn(
+                "Config-based cost overwrites is deprecated. Use external file instead (by default 'data/custom_costs.csv').",
+                DeprecationWarning,
+            )
+            logger.info(f"Overwriting {attr} with:\n{overwrites}")
+
+    return costs
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from _helpers import mock_snakemake
+
+        snakemake = mock_snakemake("build_cost_data", planning_horizons=2030)
+
+    config = snakemake.config["costs"]
+
+    snapshots = get_snapshots(
+        snakemake.params.snapshots, snakemake.params.drop_leap_day, tz="UTC"
+    )
+    nyears = len(snapshots) / 8760
+    planning_horizon = str(snakemake.wildcards.planning_horizons)
+
+    # Retrieve costs assumptions
+    # If the index is the unique pair (technology, parameter),
+    # it can be easily overwritten.
+    costs = pd.read_csv(snakemake.input.costs, index_col=[0, 1])
+    if snakemake.input.costs is not None:
+        custom_costs = pd.read_csv(
+            snakemake.params.costs["custom_cost_fn"],
```

Comment:
See other comment: Needs to be a `snakemake.input["custom_costs"]`, not a `snakemake.params`.